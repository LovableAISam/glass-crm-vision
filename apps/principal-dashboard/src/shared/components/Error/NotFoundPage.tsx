// Core
import * as React from 'react';
import Image from 'next/image';
import LogoImage from 'asset/images/logo.svg';

// Components
import {
  Typography,
  Grid,
  Stack,
} from '@mui/material';
import { Token } from '@woi/web-component';
import NotFoundImage from 'asset/images/not-found-co.svg';
import { useTranslation } from 'react-i18next';

function NotFoundPage() {
  const { t: tCommon } = useTranslation('common');

  return (
    <Grid container component="main" sx={{ height: '70vh' }}>
      <Stack justifyContent="center" alignItems="center" sx={{ width: '100%', height: '100%' }}>
        <Stack direction="row" spacing={3} alignItems="center" sx={{ mb: 10 }}>
          <Image src={LogoImage} width={50} height={50} />
          <Typography variant="h3">{tCommon('appName')}</Typography>
        </Stack>
        <Image src={NotFoundImage} />
        <Stack spacing={1} sx={{ mt: 10 }}>
          <Typography variant="h3" align="center" sx={{ mb: 1 }}>{'404. Not Found!'}</Typography>
          <Typography variant="body1" align="center" color={Token.color.greyscaleGreyDarkest}>
            {'Hi, This page not found. please contact our CS!'}
          </Typography>
        </Stack>
      </Stack>
    </Grid>
  );
}

export default NotFoundPage;