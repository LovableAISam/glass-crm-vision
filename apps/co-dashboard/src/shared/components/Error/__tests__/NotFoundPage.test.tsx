import React from 'react';
import { render, screen } from '@testing-library/react';

import NotFoundPage from '../NotFoundPage';

describe('Account rule management', () => {
  it('renders a text title text', () => {
    render(
      <NotFoundPage />
    );

    const text = screen.getByText('404. Not Found!');
    expect(text).toBeInTheDocument();
    expect(text).toMatchSnapshot();
  });

  it('renders a text description text', () => {
    render(
      <NotFoundPage />
    );

    const text = screen.getByText('Hi, This page not found. please contact our CS!');
    expect(text).toBeInTheDocument();
    expect(text).toMatchSnapshot();
  });
});