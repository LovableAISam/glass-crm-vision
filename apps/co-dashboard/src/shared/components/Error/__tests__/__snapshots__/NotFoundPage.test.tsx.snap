// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account rule management renders a text description text 1`] = `
<p
  class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter css-dsge4b-MuiTypography-root"
>
  Hi, This page not found. please contact our CS!
</p>
`;

exports[`Account rule management renders a text title text 1`] = `
<h3
  class="MuiTypography-root MuiTypography-h3 MuiTypography-alignCenter css-1nt9rxw-MuiTypography-root"
>
  404. Not Found!
</h3>
`;
